### ISSUE TITLE
error token/spans (spacy/tokens/doc.cpp:4890)

### ISSUE THREAD

**Issue by ahelaatabi** [explosion/spaCy#450]
Hey,
I came across this error :
`if token.nbor(i=1) :
  File "spacy/tokens/token.pyx", line 94, in spacy.tokens.token.Token.nbor (spacy/tokens/token.cpp:4699)
    return self.doc[self.i+i]
  File "spacy/tokens/doc.pyx", line 100, in spacy.tokens.doc.Doc.__getitem__ (spacy/tokens/doc.cpp:4890)
    if self._py_tokens[i] is not None:
IndexError: list index out of range`

What I was trying to do is run this small:
`for sent in doc.sents:
    for token in sent:
        if token.nbor(i=1) :
                print token,token.nbor(i=1)`

The purpose of this small code was to merge in the "doc" span the consecutive nouns <the ones with tag have 'NN'>. All this before passing it to a relation extraction algorithm.

Thanks in advance,

**Comment by ahelaatabi** [comment]
I've kind of managed to get pass that by limiting my code to the first Noun phrases.. The problem is that using : 
`nn=doc[list(doc).index(token):list(doc).index(token)+1]
doc.merge( token.idx , token.nbor(i = 1).idx + len(token.nbor(i = 1)) , nn.root.tag , nn.text , nn.ent_type_)`
Even the merge doesn't have any effect on the doc object itself

Would appreciate any indications to solve this

**Comment by sadovnychyi** [comment]
Same here. This code from the docs seems to be broken.

```python
for ent in doc.ents:
    ent.merge(ent.root.tag_, ent.text, ent.label_)
for np in doc.noun_chunks:
    while len(np) > 1 and np[0].dep_ not in ('advmod', 'amod', 'compound'):
        np = np[1:]
    np.merge(np.root.tag_, np.text, np.root.ent_type_)
```

**Comment by cyuxlif** [comment]
same problem

**Comment by sadovnychyi** [comment]
You can actually use `list(doc.noun_chunks)` as a temporary workaround.

**Comment by honnibal** [comment]
This should be fixed in 1.0.
